### ISSUE TITLE
Restore Hierarchy now goes to Unstack dialog 

### ISSUE THREAD

**Issue by maxwellfundi** [africanmathsinitiative/R-Instat#4338]
After discussion about the Restore Hierarchy  dialog, we have now decided that we will have this feature into the current unstack dialog. 

It will now have three options at the top(our usual radio buttons)

1. Single - Which will be unstacking a single column
2. Multiple - which will be unstacking multiple columns hence the column to unstack becomes a multiple receiver
3. Restore Hierachy - which will have no columns to unstack

**Comment by rdstern** [comment]
There are 2 separate issues here.  The issue started with the restore hierarchy, which will be the third option here.  This seems to be quite easy, i.e. I think it can (almost) be done already!  To me that is the one we need first.  Hence I suggest we work on that, while we (temporarily at least) leave the multiple option (in the top radio buttons disabled.

I took a dataset called mintemp - that was initially described in #4278.  I'll attach it.  There the first column is the ID column and is a factor.  The next 4 columns are at the same "level", i.e. name, lat, long and altitude.  So, the unstacked data has 54 rows (one for each station) and 5 columns.

If you use the current unstack and (for now) choose to unstack any column you like, and add the 4 carry columns, and then use the script you get the main command as:
# code generated by the dialog Unstack Columns

Mintemp <- InstatDataObject$get_data_frame(data_name="Mintemp")
Mintemp_stacked <- reshape2::dcast(data=Mintemp, **value.var="dup1"**, formula=**id** + station_name + longitude + latitude + elevation ~ id)
InstatDataObject$import_data(data_tables=list(Mintemp_stacked=**Mintemp_stacked**))

rm(list=c("Mintemp", "Mintemp_stacked"))

1. Now, with the third option, you won't have the **value.var="dup1"** there, because there are no columns to unstack, **so get rid of that**.  The dcast command still runs fine.  Try.  It produces the unstacked columns however with zeros and a single number in each unstacked column, giving the count. 
2. Now, you don't want those unstacked columns.  So, one way is to change the **Mintemp_stacked** into Mintemp_stacked[c(1:5)].  The 5 there is the number of columns you are producing.  That will have to be worked out.
3. There is one further change.  In the ordinary unstack the main (factor) column - here it is the id column, is not in the unstacked group, because it is the column names for the unstacked data.  You can always add it to the carry columns (as I did in the command above), but it would be neater if the code did that for you.  It will usually be the first one as I have done here.      

I hope this is easy to do.  How neat.

4. Finally, what happens if there are mistakes in the data?  I checked this by changing one of the latitude numbers (from 29.2 to 30.2).  In that case the unstack created a new factor level in the unstacked data.  So there were 54 levels in the ID columns and 55 in the unstacked data.  That's fine.  We don't need any further checks in my view.  This is a common problem where we thought there were 54 stations, but the data has created a new one and that is something we must look out for!   If necessary we could have a checkbox with the label perhaps "Keep all columns" and tooltip, "To diagnose problems in the data".  If that is checked, then we get the 54 columns (one for each level of the ID).  If there is no problem, then there is a single non-zero value in each column, giving the count at that station.  If there is a problem at any station, then there are multiple non-zero values.

Good stuff this dplyr!    
[Mintemp for restore hierarchy.zip](https://github.com/africanmathsinitiative/R-Instat/files/1592811/Mintemp.for.restore.hierarchy.zip)

**Comment by maxwellfundi** [comment]
@Muthenya  please have a look at the Unstack dialog and make the designer changes described on this issue. 

Thanks.

**Comment by rdstern** [comment]
What is the current state of this feature?  It would be good to have it in Version 4.12?

**Comment by Muthenya** [comment]
@rdstern most of the work has already been done. What is remaining concerns the issue of including a checkbox and a tooltip. Maybe that can be discussed more and also the Multiple option.

**Comment by rdstern** [comment]
That's great.  I would argue that the Multiple would be useful, but is of lower priority.  Once the restore hierarchy is done, then it would be good to get it merged.

**Comment by maxwellfundi** [comment]
@Muthenya  I thought this was fixed. Can this be closed now?
